src/solidlsp/language_servers/common.py

```diff
@@ -104,3 +122,91 @@ def _install_from_url(dep: RuntimeDependency, logger: LanguageServerLogger, targ
             FileUtils.download_and_extract_archive(logger, dep.url, dest, dep.archive_type)
         else:
             FileUtils.download_and_extract_archive(logger, dep.url, target_dir, dep.archive_type or "zip")
+
+
+class NodeJsUtils:
+    """
+    Utility functions for Node.js executable resolution across all operating systems.
+    """
+
+    @staticmethod
+    def find_node_executable() -> str | None:
+        """
+        Find the path to the node executable.
+        Returns the full path to node executable or None if not found.
+        """
+        return shutil.which("node")
+
+    @staticmethod
+    def find_npm_executable() -> str | None:
+        """
+        Find the path to the npm executable.
+        Returns the full path to npm executable or None if not found.
+        """
+        return shutil.which("npm")
+
+    @staticmethod
+    def get_npm_cli_script_path() -> str | None:
```

Comment:
I don't see a need for this class, all functionality can be in a top level function

`def get_npm_cli_script_path`

If we want to have a class serving as namespace for this, we should rather have 

```python

class CommandUitls:
   # the only method for now
    def get_npm_path

```

Public methods form fart of the interface, and should be kept to a minimum. 

Everything else that this class is doing should be done by the caller, or within that method (no point in wrapping shutil.which in a separate method which is not required as parto of an interface)

Reply:
replaced with `CommandUtils`